### DISCUSSION TITLE
Migration Assistant

### DISCUSSION THREAD

**Discussion by jeffbcross** [nrwl/nx-console#1045] [Feature Requests]
It'd be nice to have a custom UI for migrate. I can never remember the order of steps required to run migrate. In general the process feels very opaque, and makes me dread running migrations.

Some Ideas

- Show me which remote packages are eligible for updating (or even better, notify me when an update is available, to encourage me to keep my workspace evergreen)
- Show me migrations that would be run if I choose to (so I can determine how much effort the migration will require)
- Show me what files would be affected from running migrations (even better: show me a diff of changes)
- Allow me to choose from available remote versions of packages, to allow me to migrate in steps
- Run the migration and show me what succeeded/failed

**Comment by sandikbarr** [comment]
Adding on to this that the `nx migrate` command under the Common Nx Commands section does not currently accept or build any options or the package or version into the command. It just runs `nx migrate` which fails with the error `Provide the correct package name and version. E.g., @nrwl/workspace@9.0.0`
